### ISSUE TITLE
EMS actuator states not initialized soon enough for sizing

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5710]
## Issue overview

User noticed that sizing results changed dramatically (e.g. zone design heating load changed from 27,000 to zero in one zone) between runs.  The only difference was that this program had been added which just sets a global variable to be used as a custom output:

```
  EnergyManagementSystem:ProgramCallingManager,
    Total Bldg Load EMS,     !- Name
    EndOfSystemTimestepBeforeHVACReporting,  !- EnergyPlus Model Calling Point
    AEDG_BldgLoad;           !- Program Name 1
  EnergyManagementSystem:Program,
    AEDG_BldgLoad,           !- Name
    SET BldgHeating = PSZ1_Heating + PSZ2_Heating + PSZ3_Heating + PSZ4_Heating,  !- Program Line 1
    SET BldgCooling = PSZ1_Cooling + PSZ2_Cooling + PSZ3_Cooling + PSZ4_Cooling;  !- Program Line 2
```

Other EMS programs in the same file are called at InsideHVACSystemIterationLoop and set actuators like zone thermostat setpoints and infiltration rates.  It appears that these actuators get changed from Null when the  AEDG_BldgLoad program runs and this impacts the sizing simulation.  Adding a bunch of assignements to null inside AEDG_BldgLoad results in the same sizing in both runs.

NoSizingProblem_V85-Fixed-NoVerboseDebug - the base file without the AEDG_BldgLoad program
SizingProblem_V85-Fixed.idf - the problem file with the AEDG_BldgLoad program snippet and strange sizing results
SizingProblem_V85-Fixed2.idf - the fixed file with extra null statements, sizing results match the base file

Note that there are other EMS errors that will cause a fatal error, but that does not happen until after sizing is done, so you can still see the sizing results in the eio output.
### Details

Some additional details for this issue (if relevant):
- Platform (Win64)
- Version of EnergyPlus (v8.5.0 release and current state of the branch on PR #5660 8.5.0-82da0bc86b)
- Helpdesk ticket number 11400
### Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Defect file added (DefectFiles\5710)
- [x] Ticket added to Pivotal for defect (development team task)
- [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by EnergyArchmage** [comment]
Somehow the thermostat setpoint is coming out to be 0.0 during zone sizing. 

   *\* Severe  *\* UpdateZoneSizing: Cooling supply air temperature (calculated) within 2C of zone temperature
   *\*   ~~~   *\* ...check zone thermostat set point and design supply air temperatures
   *\*   ~~~   *\* ...zone name = BACK_SPACE
   *\*   ~~~   *\* ...design sensible cooling load = 852423.67 W
   *\*   ~~~   *\* ...thermostat set point temp    = 0.000 C

It looks like the Erl variable for the actuator is not getting initialized until after sizing is done.  

There is also a separate input problem in the defect file here in that there are at least four schedules ("Fans D1" "Fans D2" etc.) that have both sensors and actuators on them, this is a circular problem.  (When you need a sensor on a schedule that is also actuated the sensor should go on a copy of the schedule.  )

**Comment by EnergyArchmage** [comment]
The addition of a new and different calling point changes when the EMS input is gotten.  
 Calling point InsideHVACSystemIterationLoop  does not ever get called during sizing but EndOfSystemTimestepBeforeHVACReporting does.  

What was happening is that the EMS actuators are created and the new Erl variables associated with them start out as a value type number.  

```
  ErlVariable( VariableNum ).Value.Type = ValueNumber;
```

But the Erl variables for actuators should start out as ValueNull.  The actuators variables do get initialized to ValueNull later when EMS is more complete but not before sizing.    During sizing the model is not complete and EMS is far from done setting up and processing input. ManageSizing does not have all the same calling points as regular simulation.

So fix seems to be nulling out the Erl variable for the actuator right away during get input for the actuators.  This seems to work, see branch:
https://github.com/NREL/EnergyPlus/compare/5710-EMS-Actuator-Init

**Comment by Myoldmopar** [comment]
Closed via #5792
